### PR TITLE
feat: Add GCS backend

### PR DESCRIPTION
**Pull request by mi-char**
I am adding support for Google Cloud Storage.

### CODE REVIEW

File: gcs/src/main/scala/com/avast/clients/storage/gcs/GcsStorageBackend.scala [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+  }
+}
+
+object GcsStorageBackend {
+  def fromConfig[F[_]: ConcurrentEffect: ContextShift](
+      config: Config,
+      blocker: Blocker): Either[ConfigurationException, Resource[F, GcsStorageBackend[F]]] = {
+
+    def composeConfig: Either[ConfigurationException, GcsBackendConfiguration] = {
+      val DefaultConfig = ConfigFactory.defaultReference().getConfig("gcsBackendDefaults")
```
Comment (jendakol) [inline gcs/src/main/scala/com/avast/clients/storage/gcs/GcsStorageBackend.scala:121]: this belongs to private final val inside of the object

**Review by jendakol — CHANGES_REQUESTED** [review]

File: gcs/src/main/scala/com/avast/clients/storage/gcs/GcsStorageBackend.scala [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+import pureconfig.error.ConfigReaderException
+import pureconfig.generic.ProductHint
+import pureconfig.generic.auto._
+import pureconfig.{CamelCase, ConfigFieldMapping}
+
+import java.io.FileInputStream
+import java.nio.file.StandardOpenOption
+import java.security.{DigestOutputStream, MessageDigest}
+
+class GcsStorageBackend[F[_]: Sync: ContextShift](bucket: Bucket)(blocker: Blocker)(implicit F: Async[F])
```
[the remark this answers is not in the document]
Comment (mi-char) [reply gcs/src/main/scala/com/avast/clients/storage/gcs/GcsStorageBackend.scala]: Sync should be enough, changed.

File: gcs/src/main/scala/com/avast/clients/storage/gcs/GcsStorageBackend.scala [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+      .handleError {
+        case e: GcStorageException =>
+          logger.error(s"Error while checking presence of file $sha256 in GCS", e)
+          Either.left[StorageException, HeadResult] {
+            StorageException.InvalidResponseException(e.getCode, e.getMessage, e.getReason)
+          }
+      }
+  }
+
+  override def get(sha256: Sha256, dest: File): F[Either[StorageException, GetResult]] = {
```
[the remark this answers is not in the document]
Comment (mi-char) [reply gcs/src/main/scala/com/avast/clients/storage/gcs/GcsStorageBackend.scala:48]: Fixed.

File: gcs/src/main/scala/com/avast/clients/storage/gcs/GcsStorageBackend.scala [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+          StorageException.InvalidResponseException(e.getCode, e.getMessage, e.getReason)
+        }
+    }
+  }
+
+  private def receiveStreamedFile(blob: Blob, destination: File, expectedHash: Sha256): F[Either[StorageException, GetResult]] = {
+    logger.debug(s"Downloading streamed data to $destination")
+
+    val openOptions = Seq(StandardOpenOption.WRITE, StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING)
+
```
[the remark this answers is not in the document]
Comment (mi-char) [reply gcs/src/main/scala/com/avast/clients/storage/gcs/GcsStorageBackend.scala]: Fixed.

File: gcs/src/main/scala/com/avast/clients/storage/gcs/GcsStorageBackend.scala (hunk 1 above)
Comment (mi-char) [reply gcs/src/main/scala/com/avast/clients/storage/gcs/GcsStorageBackend.scala]: Changed, but the original code was copy-pasted from your HCP backend implementation 😉

File: gcs/README.md [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import com.avast.clients.storage.gcs.GcsStorageBackend
+import com.typesafe.config.Config
+import monix.eval.Task
+import monix.execution.Scheduler
+
+implicit val scheduler: Scheduler = ???
+val config: Config = ???
+
+def initClient: Task[GcsStorageBackend[Task]] = {
+  GcsStorageBackend.fromConfig[Task](config) match {
```
[the remark this answers is not in the document]
Comment (mi-char) [reply gcs/README.md]: The whole example has been re-created.

**Review by jendakol — APPROVED** [review]

**Pull request merged**